File: leetcode/medium/1381. Design a Stack With Increment Operation/solution.js
// https://leetcode.com/problems/design-a-stack-with-increment-operation/
class CustomStack {
    /**
     * @param {number} maxSize 
     */
    constructor(maxSize) {
        this.size = maxSize;
        this.stack = [];
    }

    /**
     *  @param {number} x
     *  @return {void}
     */
    push(x) {
        if (this.stack.length >= this.size) return;
        this.stack.push(x);
    }

    /**
     * @return {number}
     */
    pop() {
        if (this.stack.length === 0) return -1;
        return this.stack.pop();
    }

    /**
     * @param {number} k 
     * @param {number} val 
     * @return {void}
     */
    increment(k, val) {
        if (this.stack.length < k) k = this.stack.length;
        for (let i = 0; i < k; i += 1) {
            this.stack[i] += val;
        }
    }
}

File: leetcode/medium/1381. Design a Stack With Increment Operation/solution.cc
#include <vector>
#include <algorithm>

using namespace std;

class CustomStack {
public:
    int size = 0;
    vector<int> stack;

    CustomStack(int maxSize) {
        size = maxSize;
    }
    
    void push(int x) {
        if (stack.size() == size) return;
        stack.push_back(x);
    }
    
    int pop() {
        if (stack.size() == 0) return -1;
        int last = stack[stack.size() - 1];
        stack.pop_back();
        return last;
    }
    
    void increment(int k, int val) {
        int m = min(k, (int)stack.size());
        for (int i = 0; i < m; i += 1) {
            stack[i] += val;
        }
    }
};

int main() {
    return 0;
}